src/plots/plots.js

```diff
@@ -2099,6 +2085,26 @@ plots.doCalcdata = function(gd, traces) {
     }
 };
 
+plots.initCategories = function(axList) {
+    var hasCategoryAxis = false;
+
+    // initialize the category list, if there is one, so we start over
+    // to be filled in later by ax.d2c
+    for(var i = 0; i < axList.length; i++) {
+        axList[i]._categories = axList[i]._initialCategories.slice();
```

Comment:
Certainly would have been caught by now if this could ever be undefined, right? (I'm too cautious about checking for undefined unnecessarily)